code/modules/wiremod/shell/module.dm

```diff
@@ -38,38 +58,82 @@
 		return
 	if(!shell.attached_circuit)
 		return
-	var/list/action_components = shell.attached_circuit.get_all_contents_type(/obj/item/circuit_component/equipment_action/mod)
-	if(!action_components.len)
-		shell.attached_circuit.interact(mod.wearer)
+	shell.attached_circuit?.interact(mod.wearer)
+
+/obj/item/mod/module/circuit/get_configuration(mob/user)
+	. = ..()
+	for(var/obj/item/circuit_component/equipment_action/action_comp in action_comps)
+		.[REF(action_comp)] = add_ui_configuration(action_comp.button_name.value, "pin", !!action_comp.granted_to[REF(user)])
+
+/obj/item/mod/module/circuit/configure_edit(key, value)
+	. = ..()
+	var/obj/item/circuit_component/equipment_action/action_comp = locate(key)
+	if(!istype(action_comp))
```

Comment:
make sure this is sanitized in some way
If you can guarantee that the component needs to be in some list, consider doing `locate(key) in list_of_action_comps`